scalr/resource_scalr_event_bridge_integration.go

```diff
@@ -0,0 +1,132 @@
+				Description:      "AWS account ID.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+			},
+			"region": {
+				Description:      "AWS region.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+			},
+			"event_source_name": {
+				Description: "Event source name.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"event_source_arn": {
+				Description: "ARN of the event source.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceScalrEventBridgeIntegrationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	scalrClient := meta.(*scalr.Client)
+	// Get attributes.чи щось піде не так
+	name := d.Get("name").(string)
+	awsAccountID := d.Get("aws_account_id").(string)
+	region := d.Get("region").(string)
+
+	options := scalr.EventBridgeIntegrationCreateOptions{
+		Name:         &name,
+		AWSAccountId: &awsAccountID,
+		Region:       &region,
+	}
+
+	log.Printf("[DEBUG] Create EventBridge integration: %s", name)
+	integration, err := scalrClient.EventBridgeIntegrations.Create(ctx, options)
+	if err != nil {
+		return diag.Errorf("Error creating EventBridge integration %s: %v", name, err)
+	}
+	d.SetId(integration.ID)
+
+	return resourceScalrEventBridgeIntegrationRead(ctx, d, meta)
+}
+
+func resourceScalrEventBridgeIntegrationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	scalrClient := meta.(*scalr.Client)
+	integrationID := d.Id()
+
+	log.Printf("[DEBUG] Read EventBridge integration with ID: %s", integrationID)
+	EventBridgeIntegration, err := scalrClient.EventBridgeIntegrations.Read(ctx, integrationID)
+	if err != nil {
+		log.Printf("[DEBUG] EventBridge integration %s no longer exists", integrationID)
+		d.SetId("")
+		return nil
+	}
+	_ = d.Set("name", EventBridgeIntegration.Name)
+	_ = d.Set("aws_account_id", EventBridgeIntegration.AWSAccountId)
+	_ = d.Set("region", EventBridgeIntegration.Region)
+	_ = d.Set("event_source_name", EventBridgeIntegration.EventSource)
+	_ = d.Set("event_source_arn", EventBridgeIntegration.EventSourceARN)
+
+	return nil
+}
+
+func resourceScalrEventBridgeIntegrationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if d.HasChange("name") {
+		return diag.Errorf("Error updating EventBridge integration %s, name cannot be changed.", d.Id())
+	}
+
+	if d.HasChange("aws_account_id") {
+		return diag.Errorf("Error updating EventBridge integration %s, AWS account ID cannot be changed.", d.Id())
+	}
+
+	if d.HasChange("region") {
+		return diag.Errorf("Error updating EventBridge integration %s, AWS account ID cannot be changed.", d.Id())
+	}
+
+	return nil
```

Comment:
So basically we can't update the EventBridge integration.
All these fields are marked as `ForceNew` anyway, so do we need this function at all? BTW it is not connected as `UpdateContext` function in resource schema.